golang/lib/services/service.go

```diff
@@ -21,3 +21,24 @@ package services
 The identifier used for services with the network.
 */
 type ServiceID string
+
+
+type SharedDirectory struct {
```

Comment:
I'd recommend adding a really explanatory comment what this thing is, since it's a bit confusing. Basic points that would be good to touch on: 1) there's a directory shared between this code and inside the service 2) anything you put in that directory will show up on the service, and vice versa 3) recommendations/examples on how to use this object (since I bet it'll be confusing at first)

Reply:
PS prob worth putting this into its own file